porcupine/plugins/statusbar.py

```diff
@@ -1,67 +1,192 @@
 
-    def show_cursor_or_selection(self, junk: object = None) -> None:
         try:
             # For line count, if the cursor is in beginning of line, don't count that as another line.
-            chars = count(self.tab.textwidget, "sel.first", "sel.last")
-            lines = count(self.tab.textwidget, "sel.first", "sel.last - 1 char", option="-lines")
+            chars = count(self._tab.textwidget, "sel.first", "sel.last")
+            lines = count(self._tab.textwidget, "sel.first", "sel.last - 1 char", option="-lines")
         except tkinter.TclError:
             # no text selected
-            line, column = self.tab.textwidget.index("insert").split(".")
-            self.right_label.config(text=f"Line {line}, column {column}")
+            line, column = self._tab.textwidget.index("insert").split(".")
+            self.selection_label.config(text=f"Line {line}, column {column}")
         else:
-            if lines == 0:
-                self.right_label.config(text=f"{chars} characters selected")
+            if chars == 1:
+                char = self._tab.textwidget.get("sel.first")
+                hex_codepoint = hex(ord(char))[2:]
+                if ord(char) in range(128):
+                    # Non-hex also displayed in case you are debugging c code with printf
+                    text = f"ASCII character {ord(char)} (hex {hex_codepoint})"
+                else:
+                    text = f"Unicode character U+{hex_codepoint.upper()}: {unicodedata.name(char)}"
+            elif lines == 0:
+                text = f"{chars} characters selected"
             else:
-                self.right_label.config(text=f"{chars} characters on {lines+1} lines selected")
+                text = f"{chars} characters on {lines+1} lines selected"
+            self.selection_label.config(text=text)
+
+        self._encoding_button.config(text=self._tab.settings.get("encoding", str))
+        self._line_ending_button.config(
+            text=self._tab.settings.get("line_ending", settings.LineEnding).name
+        )
 
     def show_reload_warning(self, event: utils.EventWithData) -> None:
         if event.data_class(tabs.ReloadInfo).had_unsaved_changes:
             oops = utils.get_binding("<<Undo>>")
-            self.left_label.config(
-                foreground="red",
-                text=(
-                    f"File was reloaded with unsaved changes. Press {oops} to get your changes"
-                    " back."
-                ),
-            )
+            text = f"File was reloaded with unsaved changes. Press {oops} to get your changes back."
+            self.path_label.config(foreground="red", text=text)
 
     def clear_reload_warning(self, junk: object) -> None:
-        if self.left_label["foreground"]:
-            self.left_label.config(foreground="")
-            self.show_path()
+        if self.path_label["foreground"]:
+            self.path_label.config(foreground="")
+            self.update_labels()
```

Comment:
Check unnecessary, update_labels won't ruin it

```suggestion
        self.path_label.config(foreground="")
        self.update_labels()
```